packages/react/src/components/CheckboxGroup/CheckboxGroup.tsx

```diff
@@ -71,9 +71,11 @@ const CheckboxGroup = ({
 
   const [checkedNames, dispatch] = useReducer(reducer, checkedItems(items));
 
-  useEffect(
+  const itemsAsString = JSON.stringify(items);
```

Comment:
Ideally you want to wrap const values like this in a `useCallback` to avoid it being run on every re-render (no that stringify is slow). 

Not sure how this will perform with the deps change check in `useCallback` though since you are mutating existing items 🤔 

Suggest maybe giving it a try with `useCallback`?

Reply:
After implementing `useCallback` in this way, i was able to reproduce the bug. After removing the callback again i was no longer able to reproduce the issue.
![image](https://user-images.githubusercontent.com/32294735/222193576-39fbd37e-5287-43af-bd53-7d7fa886b50d.png)

Reply:
Yeah, I had a feel I wouldn't work as it uses the same method of comparing if its dependencies has changed as useEffect, which is a shallow comparison. An array with objects needs a "deepEquals" and fastest way for that is stringify. Thanks for testing tho!